ecosystem/typescript/sdk/src/plugins/aptos_token.ts

```diff
@@ -0,0 +1,490 @@
+import { AnyNumber } from "../bcs";
+import { MAX_U64_BIG_INT } from "../bcs/consts";
+import { Provider } from "../providers";
+import { AptosClient, OptionalTransactionArgs } from "../providers/aptos_client";
+import { TransactionBuilderRemoteABI } from "../transaction_builder";
+import { HexString, MaybeHexString } from "../utils";
+import { getPropertyValueRaw, getSinglePropertyValueRaw } from "../utils/property_map_serde";
+
+export interface CreateCollectionOptions {
+  royaltyNumerator?: number;
+  royaltyDenominator?: number;
+  mutableDescription?: boolean;
+  mutableRoyalty?: boolean;
+  mutableURI?: boolean;
+  mutableTokenDescription?: boolean;
+  mutableTokenName?: boolean;
+  mutableTokenProperties?: boolean;
+  mutableTokenURI?: boolean;
+  tokensBurnableByCreator?: boolean;
+  tokensFreezableByCreator?: boolean;
+}
+
+const PropertyTypeMap = {
+  BOOLEAN: "bool",
+  U8: "u8",
+  U16: "u16",
+  U32: "u32",
+  U64: "u64",
+  U128: "u128",
+  U256: "u256",
+  ADDRESS: "address",
+  VECTOR: "vector<u8>",
+  STRING: "string",
+};
+
+export type PropertyType = keyof typeof PropertyTypeMap;
+
+/**
+ * Class for managing aptos_token
+ */
+export class AptosToken {
+  readonly provider: Provider;
+
+  private readonly tokenType: string = "0x4::token::Token";
+
+  /**
+   * Creates new AptosToken instance
+   *
+   * @param provider Provider instance
+   */
+  constructor(provider: Provider) {
+    this.provider = provider;
+  }
+
+  private async submitTransaction(
+    account: AptosAccount,
+    funcName: string,
+    typeArgs: string[],
+    args: any[],
+    extraArgs?: OptionalTransactionArgs,
+  ) {
+    const builder = new TransactionBuilderRemoteABI(this.provider, {
+      sender: account.address(),
+      ...extraArgs,
+    });
+    const rawTxn = await builder.build(`0x4::aptos_token::${funcName}`, typeArgs, args);
+    const bcsTxn = AptosClient.generateBCSTransaction(account, rawTxn);
+    const pendingTransaction = await this.provider.submitSignedBCSTransaction(bcsTxn);
+    return pendingTransaction.hash;
+  }
+
+  /**
+   * Creates a new collection within the specified account
+   *
+   * @param creator AptosAccount where collection will be created
+   * @param description Collection description
+   * @param name Collection name
+   * @param uri URL to additional info about collection
+   * @param options CreateCollectionOptions type. By default all values set to `true` or `0`
+   * @returns The hash of the transaction submitted to the API
+   */
+  async createCollection(
+    creator: AptosAccount,
+    description: string,
+    name: string,
+    uri: string,
+    maxSupply: AnyNumber = MAX_U64_BIG_INT,
+    options?: CreateCollectionOptions,
+    extraArgs?: OptionalTransactionArgs,
+  ): Promise<string> {
+    return this.submitTransaction(
+      creator,
+      "create_collection",
+      [],
+      [
+        description,
+        maxSupply,
+        name,
+        uri,
+        options?.mutableDescription || true,
```

Comment:
|| true doesn't seem right to me. Wouldn't it be true even if mutableDescription is false? I think this should be ?? true so it becomes true only if mutableDescription is not specified